libqtile/backend/x11/core.py

```diff
@@ -146,6 +146,14 @@ def __init__(self, display_name: str = None) -> None:
         self._numlock_mask = xcbq.ModMasks.get(self.conn.get_modifier(numlock_code), 0)
         self._valid_mask = ~(self._numlock_mask | xcbq.ModMasks["lock"])
 
+    def finalize(self) -> None:
+        self.conn.conn.core.DeletePropertyChecked(
+            self._root.wid,
+            self.conn.atoms["_NET_SUPPORTING_WM_CHECK"],
+        ).check()
+        self.qtile = None
+        self.conn.finalize()
```

Comment:
Don't we need to call `remove_listener` here?

Reply:
no, the listener logic and finalization are now separate. We call remove_listener elsewhere.

Reply:
(basically, we don't call setup_listener in init because we don't want to require an eventloop whenever we create a core. Because setup_listener is not called in init, it's not a good idea to call it during finalization)